Marlin/src/gcode/host/M876.cpp

```diff
@@ -32,8 +35,8 @@
  * M876: Handle Prompt Response
  */
 void GcodeSuite::M876() {
-
-  if (parser.seenval('S')) hostui.handle_response((uint8_t)parser.value_int());
+  if(TERN1(EMERGENCY_PARSER, emergency_parser.isEnabled()))
+    if (parser.seenval('S')) hostui.handle_response((uint8_t)parser.value_int());
 
 }
```

Comment:
So I assume this is the command that you said was possible to mess up state if processed from both side-band and in the emergency parser:

> The last one may leave a stale value so its internal execution is blocked if the emergency parser is currently active.

The logic I see here looks backward.

Breaking this down into more verbose form, I'm interpreting it as follows:

```
#if ENABLED(EMERGENCY_PARSER)
   if (emergency_parser.isEnabled())
     if (parser.seenval('S')) hostui.handle_response((uint8_t)parser.value_int());
#else
     if (parser.seenval('S')) hostui.handle_response((uint8_t)parser.value_int());
#endif
```

Did you mean for this to be (added a `!`)?

`if(TERN1(EMERGENCY_PARSER, !emergency_parser.isEnabled()))`

Reply:
@InsanityAutomation ?